Add slither setup to the project

asep-fajar-firmansyah opened the pull request:
I have set up slither to the project, and also updated the README accordingly.

asep-fajar-firmansyah committed:
setup slither locally

haltakov commented on README.md:
```diff
@@ -77,3 +77,27 @@ The current NFT contract is deployed on the Mumbai testnet here: [0x35b5147E7499
 
 The current Token contract is deployed on the Mumbai testnet here: [0xc4C99f33F686A74a2Fe95B26Ce317708f605A9eA](https://mumbai.polygonscan.com/address/0xc4C99f33F686A74a2Fe95B26Ce317708f605A9eA)
 
+## Setup CI
```
I would just call this "Setup Slither-Analyzer" instead of "Setup CI". The CI will still work, this is an optional step if you want to check your code yourself before creating a PR.

haltakov commented on README.md:
````diff
@@ -77,3 +77,27 @@ The current NFT contract is deployed on the Mumbai testnet here: [0x35b5147E7499
+```
+sudo snap install solc
+python -m pip install --upgrade pip
+pip install slither-analyzer solc-select
````
We should create a `requirements.txt` file for this and add instructions how to setup a `venv`. SOmething like this should be enough.

Create a new virtual environment:
```
python -m venv .venv
source .venv/bin/activate
```

Install the required Python packages:
```
pip install -r requiremets.txt
```

haltakov reviewed (changes requested):
Thanks, this looks good! I requested some small changes.

asep-fajar-firmansyah committed:
simplification to set up a slither-analyzer step on README

asep-fajar-firmansyah committed:
simplification to set up a slither-analyzer step on README

asep-fajar-firmansyah committed:
Added requirements.txt to install the slither-analyzer

asep-fajar-firmansyah commented:
Hi @haltakov, I have completed the requests, please have a look :)

haltakov committed:
Fix package versions in Python requirements

Generate the requirements.txt file using pip freeze to fix the package
versions

haltakov committed:
Update .gitignore

Update .gitignore to ignore the .venv folder.

haltakov commented:
@asep-fajar-firmansyah - thanks! I fixed two small things and will merge it now :) You can take a look at my commits to understand what I changed.
